R/tm_g_distribution.R

```diff
@@ -343,9 +336,89 @@ srv_distribution <- function(id,
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
-    data_extract <- list(dist_i = dist_var, strata_i = strata_var, group_i = group_var)
 
-    selector_list <- teal.transform::data_extract_multiple_srv(data_extract, data)
+    rule_req <- function(value) {
+      if (isTRUE(input$dist_tests %in% c("Fligner-Killeen",
+                                         "t-test (two-samples, not paired)",
+                                         "F-test",
+                                         "Kolmogorov-Smirnov (two-samples)",
+                                         "one-way ANOVA"))) {
+        if (!shinyvalidate::input_provided(value))
+          "Please select stratify variable."
+      }
+    }
+    rule_dupl <- function(...) {
+      if (identical(input$dist_tests, "Fligner-Killeen")) {
+        strata <- selector_list()$strata_i()$select
+        group <- selector_list()$group_i()$select
+        if (isTRUE(strata == group))
+          "Please select different variables for strata and group."
```

Comment:
I'm not sure about this - in the example we are using filter specs:

![image](https://user-images.githubusercontent.com/15201933/210370815-32210331-f632-4e70-b8fe-840a0118faa9.png) 

maybe we should reduce what's allowed as arguments to the module?